app/models/media.rb

```diff
@@ -43,14 +43,13 @@ def jsondata(context = nil)
   def data(context = nil)
     #TODO:: change the assumsion for a one Pender result
     em_pender = self.annotations('embed').last
-    embed = em_pender.embed unless em_pender.nil?
+    embed = JSON.parse(em_pender.embed) unless em_pender.nil?
     em_u = self.annotations('embed')
     context_id = context.nil? ? nil : context.id
     em_u.reverse.each do |obj|
       if obj.context_id.to_i == context_id.to_i
```

Comment:
Instead of loading all annotations in memory and checking the context of each, I suggest to filter by context right away. The method allows that: https://github.com/meedan/check-api/blob/develop/app/models/concerns/annotation_base.rb#L41-L43. It's the second parameter of the `annotations` method.